tests/graph/test_all_connected_subgraphs.py

```diff
@@ -0,0 +1,168 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+import rustworkx
+
+
+class TestGraphAllConnectedSubgraphs(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.edges = [(0, 1), (1, 2), (2, 3), (0, 3), (0, 4), (4, 5), (4, 7), (7, 6), (5, 6)]
+        self.nodes = list(range(8))
+        self.expected_subgraph_nodes = [
```

Comment:
If you know the connected components you can use [`itertools.combination`](https://docs.python.org/3/library/itertools.html#itertools.combinations) to shorten this a little bit

Reply:
Thanks, I added the bruteforce method that can also serve as a reference for future work and generated the expected results for the tests that way.